src/lib/datepicker/datepicker.md

```diff
@@ -0,0 +1,185 @@
+The datepicker allows users to enter a date either through text input, or by choosing a date from
+the calendar. It is made up of several components and directives that work together:
+
+<!-- TODO: INSERT OVERVIEW EXAMPLE HERE -->
+
+### Current state
+Currently the datepicker is in the beginning stages and supports basic date selection functionality.
+There are many more features that will be added in future iterations, including:
+ * Support for datetimes (e.g. May 2, 2017 at 12:30pm) and month + year only (e.g. May 2017)
+ * Support for selecting and displaying date ranges
+ * Support for custom time zones
+ * Infinite scrolling through calendar months
+ * Built in support for [Moment.js](https://momentjs.com/) dates
+
+### Connecting a datepicker to an input
+A datepicker is composed of a text input and a calendar pop-up, connected via the `mdDatepicker`
+property on the text input.
+
+```html
+<input [mdDatepicker]="myDatepicker">
+<md-datepicker #myDatepicker></md-datepicker>
+```
+
+An optional datepicker toggle button is available. A toggle can be added to the example above:
+
+```html
+<input [mdDatepicker]="myDatepicker">
+<button [mdDatepickerToggle]="myDatepicker"></button>
+<md-datepicker #myDatepicker></md-datepicker>
+```
+
+This works exactly the same with an input that is part of an `<md-input-container>` and the toggle
+can easily be used as a prefix or suffix on the material input:
+
+```html
+<md-input-container>
+  <input mdInput [mdDatepicker]="myDatepicker">
+  <button mdSuffix [mdDatepickerToggle]="myDatepicker"></button>
+</md-input-container>
+<md-datepicker #myDatepicker></md-datepicker>
+```
+
+### Setting the calendar starting view
+By default the calendar will open in month view, this can be changed by setting the `startView`
+property of `md-datepicker` to `"year"`. In year view the user will see all months of the year and
+then proceed to month view after choosing a month.
+
+The month or year that the calendar opens to is determined by first checking if any date is
+currently selected, if so it will open to the month or year containing that date. Otherwise it will
+open to the month or year containing today's date. This behavior can be overridden by using the
+`startAt` property of `md-datepicker`. In this case the calendar will open to the month or year
+containing the `startAt` date. 
+
+```html
+...
+<md-datepicker startView="year" startAt="1/1/1990"></md-datepicker>
```

Comment:
As discussed: having date string literals are inherently tied to a specific locale.